modules/tinymce/src/plugins/charmap/test/ts/browser/CharmapDialogHeightTest.ts

```diff
@@ -10,30 +11,67 @@ import Theme from 'tinymce/themes/silver/Theme';
-
-  it('TBA: Search for items, dialog height should not change when fewer items returned', async () => {
-    const editor = hook.editor();
-    const body = SugarBody.body();
-    const doc = SugarDocument.getDocument();
-
-    TinyUiActions.clickOnToolbar(editor, 'button[aria-label="Special character"]');
-    await TinyUiActions.pWaitForDialog(editor);
-    await FocusTools.pTryOnSelector('Focus should start on', doc, 'input');
-
-    const tabPanel = UiFinder.findIn(body, '[role="dialog"] [role="tabpanel"]').getOrDie();
-    const oldHeight = tabPanelHeight(tabPanel);
-    const input = FocusTools.setActiveValue(doc, '.');
-    fakeEvent(input, 'input');
-    // need to wait until '.tox-collection__group' has no children
-    await UiFinder.pWaitForState('Wait for search to finish', body, '[role="dialog"] .tox-collection__group', (e) => Traverse.childNodesCount(e) === 0);
-    const newHeight = tabPanelHeight(tabPanel);
-    assert.equal(parseInt(newHeight, 10), parseInt(oldHeight, 10), 'New height and old height differ');
+  Arr.each([
+    { label: 'IFrame Editor', setup: TinyHooks.bddSetupLight },
+    { label: 'Shadow Dom Editor', setup: TinyHooks.bddSetupInShadowRoot },
+  ], (tester) => {
+    context(tester.label, () => {
+      const hook = tester.setup<Editor>({
+        plugins: 'charmap',
+        toolbar: 'charmap',
+        base_url: '/project/tinymce/js/tinymce'
+      }, [ Plugin, Theme ], true);
+
+      before(() => {
+        // Make the shadow host focusable
+        const target = TinyDom.targetElement(hook.editor());
+        SugarShadowDom.getShadowRoot(target).each((sr) => {
+          const host = SugarShadowDom.getShadowHost(sr);
+          Attribute.set(host, 'tabindex', 1);
+        });
+      });
+
+      const tabPanelHeight = (tabpanel: SugarElement<Element>) => Css.getRaw(tabpanel, 'height').getOrDie('tabpanel has no height');
+
+      const type = (doc: SugarElement<Document | ShadowRoot>, text: string) => {
+        const input = FocusTools.getFocused(doc).getOrDie() as SugarElement<HTMLInputElement>;
+        Value.set(input, Value.get(input) + text);
+        fakeEvent(input, 'input');
+      };
+
+      it('TBA: Search for items, dialog height should not change when fewer items returned', async () => {
+        const editor = hook.editor();
+        const root = SugarShadowDom.getRootNode(TinyDom.targetElement(editor));
+        const body = SugarShadowDom.getContentContainer(root);
+
+        TinyUiActions.clickOnToolbar(editor, 'button[aria-label="Special character"]');
+        await TinyUiActions.pWaitForDialog(editor);
+        await FocusTools.pTryOnSelector('Focus should start on', root, 'input');
+
+        const tabPanel = UiFinder.findIn(body, '[role="dialog"] [role="tabpanel"]').getOrDie();
+        const oldHeight = tabPanelHeight(tabPanel);
+        type(root, '.');
+        // need to wait until '.tox-collection__group' has no children
```

Comment:
This isn't new code, so I'm asking more for my understanding than as a request for the code to change — it would end up having no children because no icon names start with (contain?) a `.` character right?

Reply:
Yeah, that's my understanding of what it's doing.